ignite/pkg/cosmosgen/generate.go

```diff
@@ -104,23 +121,132 @@ func (g *generator) setup() (err error) {
 			if err != nil {
 				return err
 			}
+			includes := []string{}
+			if len(modules) > 0 {
 
+				includes, err = g.resolveIncludes(path) // For versioning issues, we do dependency/includes resolution per module
+				if err != nil {
+					return err
+				}
+			}
 			modulesInPath = ModulesInPath{
-				Path:    path,
-				Modules: modules,
+				Path: path, // Each go module
+				ModuleIncludes: ModuleIncludes{ // Has a set of sdk modules and a set of includes to build them
+					Includes: includes,
+					Modules:  modules,
+				},
 			}
 
 			if err := moduleCache.Put(cacheKey, modulesInPath); err != nil {
 				return err
 			}
 		}
 
-		g.thirdModules[modulesInPath.Path] = append(g.thirdModules[modulesInPath.Path], modulesInPath.Modules...)
+		g.thirdModules[modulesInPath.Path] = modulesInPath.ModuleIncludes
 	}
 
 	return nil
 }
 
+func (g *generator) getProtoIncludeFolders(modPath string) []string {
+	includePaths := []string{filepath.Join(modPath, g.protoDir)} // Add default protoDir and default includeDirs
+	for _, dir := range g.o.includeDirs {
+		includePaths = append(includePaths, filepath.Join(modPath, dir))
+	}
+	return includePaths
+}
+
+/*
+Leave this here for reference to improve resolution algorithm
+
+	func (g *generator) checkForProto(modpath string) (bool, error) {
+		err := filepath.WalkDir(modpath,
+			func(path string, _ fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				if filepath.Ext(path) == ".proto" {
+					return errFileFound
+				}
+				return nil
+			})
+		if err == errFileFound {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		return false, errFileNotFound
+	}
+*/
+func (g *generator) checkForBuf(modpath string) (string, error) {
+	var bufPath string
+	err := filepath.WalkDir(modpath,
+		func(path string, _ fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if filepath.Base(path) == "buf.yaml" {
+				bufPath = path
+				return errFileFound
+			}
+			return nil
+		})
+	if errors.Is(err, errFileFound) {
+		return bufPath, nil
+	}
+	if err != nil {
+		return "", err
+	}
+	return "", errFileNotFound
+}
+
+func (g *generator) generateBufIncludeFolder(modpath string) (string, error) {
+	protoPath, err := os.MkdirTemp("", "")
+	if err != nil {
+		return "", err
+	}
```

Comment:
It might be a good idea to add a cleanup function, maybe in a different issue, or we could add a TODO here to remember implementing a way to cleanup the exported files.

Another way to handle this king of cases could maybe be to have a folder in `$HOME/.ignite` that we could use to generate dynamic content that we could cleanup at the end.

Reply:
Yes, I wanted to discuss this with you guys because there's a bunch of temp folders/files to generate that 
a) need to be shared across tasks (so a simple defer won't cut it) or 
b) across invocations (like /temp/proto-sdk generation for each module in cosmos-sdk)

And a generalised solution to handle these would be ideal.

Reply:
For now I'm adding a simples solution which will remove these temporary folders in #3696